### PR TITLE
CoT via Bluetooth Serial

### PR DESCRIPTION
**Pull request by W3DMJ**
Receives CoT message through Bluetooth SPP. The messages are then published to the map

### CODE REVIEW

**Comment by W3DMJ** [comment]
@takdeveloper what is the build schedule for this project? I see a new version was created 7 days ago and I submitted a change the day.

File: atak/ATAK/app/src/main/java/com/atakmap/android/cot/bluetooth/BluetoothSerialCoTManager.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import com.atakmap.android.maps.MapView;
+
+/**
+ * Manages the actual Bluetooth channel and socket reading for CoT string received over LoRaWAN.
+ */
+public class BluetoothSerialCoTManager extends BluetoothCotManager {
+
+    private static final String TAG = "BluetoothSerialCoTManager";
+
+    private static final byte[] address = new byte[] {
```
Comment (bisgroves) [inline atak/ATAK/app/src/main/java/com/atakmap/android/cot/bluetooth/BluetoothSerialCoTManager.java:13]: Recommend that this not publish via a multicast address utilizing a TTL of 1.   Please utilize 
final CotEvent ce = CotEvent.parse(cotString);
CotMapComponent.getInternalDispatcher().dispatch(ce)

File: atak/ATAK/app/src/main/java/com/atakmap/android/cot/bluetooth/BluetoothSerialCoTReader.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        this.bgcm = bgcm;
+    }
+
+    @Override
+    public BluetoothCotManager getCotManager(MapView mapView) {
+        BluetoothDevice device = connection.getDevice();
+        String deviceName = device.getName();
+
+        // ATAK-8437 BluetoothGPSNMEAReader NPE
+        // I have never observed a case where the deviceName is null.
```
Comment (bisgroves) [inline atak/ATAK/app/src/main/java/com/atakmap/android/cot/bluetooth/BluetoothSerialCoTReader.java:35]: Thank you for pointing that out.    We have observed on some phones specifically with earlier versions of Android that the device getName call would indeed return null.

**Review by bisgroves — COMMENTED** [review]
Due to the very specific nature of this change set, it would likely better as a plugin.    If there is more devices that would benefit from this being in core - then it can be reconsidered.

**Comment by W3DMJ** [comment]
> Due to the very specific nature of this change set, it would likely better as a plugin. If there is more devices that would benefit from this being in core - then it can be reconsidered.

@bisgroves though I had considered writing a plugin for this device, most of the code is snippets taken from modules already existing in the core. I simply combined some of the Bluetooth functionality with code snippets from receiving GPS over serial. Thus I thought it could reside in the core and save users of this new "Poor Man's Blue Force Tracker" device, soon to be available commercially, the effort of having to install a plugin after downloading ATAK-CIV from Google Play.

On the other hand a plugin would provide an opportunity of being able to programmatically change the tracker's identifier in that a tactical callsign could be assigned in place of the MAC address of the device to more easily identify whom or what is being tracked.

File: atak/ATAK/app/src/main/java/com/atakmap/android/cot/bluetooth/BluetoothSerialCoTManager.java (hunk 1 above)
Comment (W3DMJ) [reply atak/ATAK/app/src/main/java/com/atakmap/android/cot/bluetooth/BluetoothSerialCoTManager.java:13]: @bisgroves I will consider this change. The idea of using multicast was based on a use case scenario. Say your are a member of a hunting party. Each android device is connected via a ZeroTier network. The trackers are using RF to communicate position. Once position data is received the receiving device converts the data to CoT and send it over Bluetooth Serial to ATAK and then ATAK repeats the received CoT over multicast. Now each of the connected devices can see the received tracker's information. Does "CotMapComponent.getInternalDispatcher().dispatch(ce)" cause an event that would rebroadcast the received CoT over the network?

**Pull request closed without merging**